### PR TITLE
Small tests added

### LINKED ISSUE #7: PCM Tests need to be revised
Tests need to be revised - in current form there is no possibility to pass them by travis (because tests need pmc.x executable)

### PR DESCRIPTION
**Pull request by mkuculyma**
Fixes #7

Summary of changes:
- Small tests added
- Plugin refactoring
- Task example updated
- default value nil changed to math.NaN()

How to verify it:
- run tests
- load plugin and create exemplary task

Testing done:
- Small tests

### CODE REVIEW

File: pcm/pcm_test.go [hunk 1]
```diff
@@ -38,17 +149,15 @@ func TestPCMPlugin(t *testing.T) {
...
-			So(err, ShouldBeNil)
-		})
-		Convey("So pcmCol should be of PCM type", func() {
-			So(pcmCol, ShouldHaveSameTypeAs, &PCM{})
+		pcm := &PCM{mutex: &sync.RWMutex{}, data: map[string]interface{}{}}
+		reader := strings.NewReader(mockCmdOut)
+		pcm.parse(reader)
+
+		Convey("So pcm should be of PCM type", func() {
+			So(pcm, ShouldHaveSameTypeAs, &PCM{})
```
Comment (lmroz) [inline pcm/pcm_test.go]: This variable is directly created at line 152, it can't have any other type.

File: pcm/pcm_test.go [hunk 2]
```diff
@@ -60,5 +169,81 @@ func TestPCMPlugin(t *testing.T) {
...
+
+		Convey("Given invalid metric namespace collect metrics", func() {
+			badMetrics := []plugin.MetricType{
+				plugin.MetricType{
+					Namespace_: core.NewNamespace("intel", "pcm", "bad"),
+				},
+			}
+			So(func() { pcm.CollectMetrics(badMetrics) }, ShouldNotPanic)
+			result, err := pcm.CollectMetrics(badMetrics)
+			So(err, ShouldBeNil)
```
Comment (lmroz) [inline pcm/pcm_test.go]: Given invalid metric namespace... err should be nil? I don't agree.

File: pcm/pcm_test.go [hunk 3]
```diff
@@ -60,5 +169,81 @@ func TestPCMPlugin(t *testing.T) {
...
+			So(len(result), ShouldEqual, 24)
+			So(err, ShouldBeNil)
+
+			m := make(map[string]interface{}, len(result))
+
+			for _, r := range result {
+				m[r.Namespace().String()] = r.Data()
+			}
+
+			So(len(m), ShouldEqual, len(refMap))
```
Comment (lmroz) [inline pcm/pcm_test.go]: `ShouldResemble` is testing for maps equivalence.

File: pcm/pcm_test.go [hunk 4]
```diff
@@ -22,13 +22,124 @@ limitations under the License.
...
-	"testing"
+	"strings"
+	"sync"
 
 	"github.com/intelsdi-x/snap/control/plugin"
 	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
+	"github.com/intelsdi-x/snap/core"
 	. "github.com/smartystreets/goconvey/convey"
+
+	"testing"
```
Comment (lmroz) [inline pcm/pcm_test.go]: This import should be at line 25 as it was before.

File: pcm/pcm_test.go [hunk 5]
```diff
@@ -22,13 +22,124 @@ limitations under the License.
...
+	"/intel/pcm/L3MISS":     0.0925,
+	"/intel/pcm/PhysIPC":    2.26,
+	"/intel/pcm/INSTnom":    0.0349,
+	"/intel/pcm/C1res%":     2.13,
+	"/intel/pcm/C3res%":     0.382,
+	"/intel/pcm/FREQ":       0.0155,
+	"/intel/pcm/L2HIT":      0.335,
+}
+
+var mockMts = []plugin.MetricType{
```
Comment (lmroz) [inline pcm/pcm_test.go]: Consider automatic generation of request list from `refMap`.

File: pcm/pcm_test.go [hunk 6]
```diff
@@ -60,5 +169,81 @@ func TestPCMPlugin(t *testing.T) {
...
+			mts, err := pcm.GetMetricTypes(plugin.ConfigType{})
+			So(err, ShouldBeNil)
+			So(len(mts), ShouldEqual, 46)
+
+			namespaces := []string{}
+			for _, m := range mts {
+				namespaces = append(namespaces, m.Namespace().String())
+			}
+
+			So(namespaces, ShouldContain, "/intel/pcm/INST")
```
Comment (lmroz) [inline pcm/pcm_test.go]: Seems like this can also be automated  using keys of `refMap`.

File: pcm/pcm.go [hunk 7]
```diff
@@ -72,6 +74,8 @@ func (p *PCM) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, erro
 		if v, ok := p.data[mts[i].Namespace().String()]; ok {
 			mts[i].Data_ = v
 			mts[i].Timestamp_ = time.Now()
+		} else {
+			mts[i].Namespace_ = nil
```
Comment (mkuculyma) [inline pcm/pcm.go]: Done

File: pcm/pcm_test.go (hunk 4 above)
Comment (mkuculyma) [inline pcm/pcm_test.go]: Done

File: pcm/pcm_test.go (hunk 1 above)
Comment (mkuculyma) [inline pcm/pcm_test.go]: Corrected

File: pcm/pcm_test.go (hunk 2 above)
Comment (mkuculyma) [inline pcm/pcm_test.go]: Corrected

File: pcm/pcm_test.go (hunk 3 above)
Comment (mkuculyma) [inline pcm/pcm_test.go]: Done

File: pcm/pcm_test.go (hunk 6 above)
Comment (mkuculyma) [inline pcm/pcm_test.go]: Done

**Pull request merged**